gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/resource/ApisResource.java

```diff
@@ -159,8 +157,12 @@ private ApiQuery createQueryFromParam(ApisParam apisParam) {
             apiQuery.setName(apisParam.getName());
             apiQuery.setTag(apisParam.getTag());
             apiQuery.setVersion(apisParam.getVersion());
-            if (apisParam.getCategory() != null) {
+
+            boolean isCategoryMode = (apisParam.getCategory() != null && apisParam.getFilter() == null);
+            if (isCategoryMode && apisParam.getCategory() != null) {
```

Comment:
the condition `apisParam.getCategory() != null` seems to be duplicate what do you think about 
```suggestion
            apisParam.setCategory(isCategoryMode ? apisParam.getCategory() : null);
```